fendermint/eth/api/src/conv/from_eth.rs

```diff
@@ -7,23 +7,17 @@ use anyhow::Context;
 use ethers_core::types::{transaction::eip2718::TypedTransaction, H256};
 
 pub use fendermint_vm_message::conv::from_eth::*;
-use fvm_shared::{error::ExitCode, message::Message};
+use fvm_shared::message::Message;
 
-use crate::{error, JsonRpcResult};
+use crate::JsonRpcResult;
 
 pub fn to_tm_hash(value: &H256) -> anyhow::Result<tendermint::Hash> {
     tendermint::Hash::try_from(value.as_bytes().to_vec())
         .context("failed to convert to Tendermint Hash")
 }
 
 pub fn to_fvm_message(tx: TypedTransaction) -> JsonRpcResult<Message> {
-    match tx {
-        TypedTransaction::Eip1559(tx) => {
-            Ok(fendermint_vm_message::conv::from_eth::to_fvm_message(&tx)?)
-        }
-        TypedTransaction::Legacy(_) | TypedTransaction::Eip2930(_) => error(
-            ExitCode::USR_ILLEGAL_ARGUMENT,
-            "unexpected transaction type",
-        ),
-    }
+    Ok(fendermint_vm_message::conv::from_eth::to_fvm_message(
+        &tx.into(),
+    )?)
```

Comment:
I'm not sure this will work during signature checks. 

This `tx.into()` calls [TypedTransaction::into_eip1559](https://github.com/gakonst/ethers-rs/blob/ethers-v2.0.8/ethers-core/src/types/transaction/eip2718.rs#L562) which doesn't set all the fields (leaves out `max_fee_per_gas` and  `max_priority_fee_per_gas` which the FVM actually uses) and leaves them on default values. Which in itself might be okay, we send the transaction with 0 fees, however, this happens _after_ the transaction has been signed by the client, so let's see how we check the signature.

To figure out what to check the signature against, we turn the FVM transaction back into a `TypedTransaction` and call [sighash](https://github.com/consensus-shipyard/fendermint/blob/master/fendermint/vm/message/src/signed.rs#L102) on it. Since we only expect Eip1559 transactions, that's exactly what we're are [going to use](https://github.com/consensus-shipyard/fendermint/blob/master/fendermint/vm/message/src/conv/from_fvm.rs#L107).

Then if we look at [TypedTransaction::sighash](https://github.com/gakonst/ethers-rs/blob/ethers-v2.0.8/ethers-core/src/types/transaction/eip2718.rs#L363) it calls [TypedTransaction::rlp](https://github.com/gakonst/ethers-rs/blob/ethers-v2.0.8/ethers-core/src/types/transaction/eip2718.rs#L337) which treats all variants differently, and in our case [prepends](https://github.com/gakonst/ethers-rs/blob/ethers-v2.0.8/ethers-core/src/types/transaction/eip2718.rs#L348) `0x2` to the RLP representation. 

Therefore if the transaction from the client _wasn't_ an Eip1559 request, you might turn it into one here, but it will fail the signature check because what has been signed by the client had a different, or no prefix.

It would be better to reject this transaction when we still understand why we are doing it.

Reply:
And I know that with the current workaround these other transactions would be rejected by the API, but if we treat this as a temporary solution, when ethers is released with the fix, the above scenario will be a possibility.

Reply:
Resolved in the latest commit.